src/CoreWCF.ConfigurationManager/src/CoreWCF/Configuration/IssuedTokenParametersElement.cs

```diff
@@ -0,0 +1,328 @@
+        }
+
+        internal void ApplyConfiguration(IssuedSecurityTokenParameters parameters)
+        {
+            if (parameters == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentNullException("parameters"));
+
+            if (this.AdditionalRequestParameters != null)
+            {
+                foreach (XmlElementElement e in this.AdditionalRequestParameters)
+                {
+                    parameters.AdditionalRequestParameters.Add(e.XmlElement);
+                }
+            }
+
+            if (this.ClaimTypeRequirements != null)
+            {
+                foreach (ClaimTypeElement c in this.ClaimTypeRequirements)
+                {
+                    parameters.ClaimTypeRequirements.Add(new ClaimTypeRequirement(c.ClaimType, c.IsOptional));
+                }
+            }
+
+            parameters.KeySize = this.KeySize;
+            //TODO: This is an internal property
+            //parameters.KeyType = this.KeyType;
+            parameters.DefaultMessageSecurityVersion = this.DefaultMessageSecurityVersion;
+            parameters.UseStrTransform = this.UseStrTransform;
+
+            if (!string.IsNullOrEmpty(this.TokenType))
+            {
+                parameters.TokenType = this.TokenType;
+            }
+            if (PropertyValueOrigin.Default != this.ElementInformation.Properties[ConfigurationStrings.Issuer].ValueOrigin)
+            {
+                //TODO: Implement IssuedTokenParameters
+                //this.Issuer.Validate();
+
+                //TODO: Implement IssuedTokenParameters
+                //parameters.IssuerAddress = ConfigLoader.LoadEndpointAddress(this.Issuer);
+
+                if (!string.IsNullOrEmpty(this.Issuer.Binding))
+                {
+                    //TODO: Implement IssuedTokenParameters
+                    //parameters.IssuerBinding = ConfigLoader.LookupBinding(this.Issuer.Binding, this.Issuer.BindingConfiguration, this.EvaluationContext);
+                }
+            }
+
+            if (PropertyValueOrigin.Default != this.ElementInformation.Properties[ConfigurationStrings.IssuerMetadata].ValueOrigin)
+            {
+                //TODO: Implement IssuedTokenParameters
+                //parameters.IssuerMetadataAddress = ConfigLoader.LoadEndpointAddress(this.IssuerMetadata);
+            }
+        }
+
+        internal void Copy(IssuedTokenParametersElement source)
+        {
+            if (this.IsReadOnly())
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ConfigurationErrorsException(SR.Format(SR.ConfigReadOnly)));
+            }
+            if (null == source)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("source");
+            }
+
+            //TODO: Implement IssuedTokenParameters
+            //foreach (XmlElementElement xmlElement in source.AdditionalRequestParameters)
+            //{
```

Comment:
If this isn't being implemented, can you throw inside the for loop, or `if (source.AdditionalParameters.Any()) { throw ....}`. Same with the ClaimTypeRequirements

Reply:
I have uncommented these